### PR TITLE
Configure dependabot.yml

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
GitHub Dependabot is adding PRs to our repo once upon a time, like #4317

By default, it creates `dependency` label and adds it to the PR. This should prevent the label from appearing.

Changes are based on [docs for dependabot](https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates) and [labels section](https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates#labels)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# https://docs.github.com/en/code-security/supply-chain-security/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    labels: [ ]     # Do not add default "dependency" label
```
Comment (costin-zaharia-sonarsource) [inline .github/dependabot.yml:10]: Can we add the `tooling` label?

File: .github/GitHub.shproj [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (costin-zaharia-sonarsource) [inline .github/GitHub.shproj:1]: This seems to be a misuse of the shared project. We can link just the yml file. Or add a pattern in the *.sln file to include in Common solution items. WDYT?

**Review by costin-zaharia-sonarsource — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply .github/dependabot.yml:10]: Yes we can, but updating dependencies for security reasons isn't tooling. More of an `improvement`, but in general, we don't put labels on standalone PRs

Comment (pavel-mikula-sonarsource) [reply .github/dependabot.yml:10]: Ah, this is in github action dependencies (not visible in UI 🙂 ), so yes, we could, be we're not putting labels on standalone PRs so we can be without it

File: .github/GitHub.shproj (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply .github/GitHub.shproj:1]: I admit it's little bit of cheating, but it doesn't get build nor referenced. I haven't found a way how to achieve this in other cost-less way - I want to see entire content of `.github` directory without a need to update SLN file everytime something changes to keep maintenance cost low.

Comment (costin-zaharia-sonarsource) [reply .github/GitHub.shproj:1]: Adding a pattern there does not help?

Comment (pavel-mikula-sonarsource) [reply .github/GitHub.shproj:1]: I couldn't find any docs for this format in SLN file. This [SO answer](https://stackoverflow.com/a/47637436/1486185) claims it as `relative path to file = relative path to file`. I don't have a guess how to inject pattern there.
```
	ProjectSection(SolutionItems) = preProject
		..\.editorconfig = ..\.editorconfig
		src\AssemblyInfo.Shared.cs = src\AssemblyInfo.Shared.cs
		..\azure-pipelines-external.yml = ..\azure-pipelines-external.yml
		..\azure-pipelines.yml = ..\azure-pipelines.yml
		..\scripts\build\build-utils.ps1 = ..\scripts\build\build-utils.ps1
		..\scripts\build\dev-build.ps1 = ..\scripts\build\dev-build.ps1
		Directory.Build.props = Directory.Build.props
		NuGet.Config = NuGet.Config
		src\Signing.cs = src\Signing.cs
		src\SonarAnalyzer.ruleset = src\SonarAnalyzer.ruleset
		tests\SonarAnalyzer.Test.ruleset = tests\SonarAnalyzer.Test.ruleset
		StyleCopAnalyzers.targets = StyleCopAnalyzers.targets
		..\scripts\utils.ps1 = ..\scripts\utils.ps1
	EndProjectSection
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**